player/lua/select.lua

```diff
@@ -345,6 +345,59 @@ mp.add_key_binding(nil, "select-audio-device", function ()
     })
 end)
 
+mp.add_key_binding(nil, "select-history", function ()
+    local watch_later_dir = mp.get_property("current-watch-later-dir")
+
+    if not watch_later_dir then
+        show_error("No watch later files found.")
+        return
+    end
+
+    local watch_later_files = {}
+
+    for i, file in ipairs(utils.readdir(watch_later_dir, "files") or {}) do
+        watch_later_files[i] = watch_later_dir .. "/" .. file
+    end
+
+    if #watch_later_files == 0 then
+        show_error("No watch later files found.")
+        return
+    end
+
+    local recent_files = {}
+    for _, watch_later_file in pairs(watch_later_files) do
+        local file_handle = io.open(watch_later_file)
+        if file_handle then
+            local line = file_handle:read()
+            if line and line ~= "# redirect entry" and line:find("^#") then
+                recent_files[#recent_files + 1] = {watch_later_file, line:sub(3)}
+            end
+            file_handle:close()
+        end
+    end
+
+    if #recent_files == 0 then
+        show_error("Enable --write-filename-in-watch-later-config to select recent files.")
```

Comment:
This will show, even if this option is enabled, but there is not recent files.

Reply:
It prints No watch later files found. and returns earlier if no recent files are saved. This is specifically if you have watch later files but did not enable `--write-filename-in-watch-later-config`.

Reply:
Yes, and it will show all the time until you watch at least two files with this option, enabled. I think it should detect if this option is enabled. Because it was confusing to me to see this error, after enabling this option.